automatic_score_report.py

```diff
@@ -0,0 +1,88 @@
+import os
+import json
+import argparse
+import numpy as np
+import pandas as pd
+
+
+def load_results(results_dir_: str):
+    results_path = os.path.join(results_dir_, 'run.json')
+    with open(results_path) as f:
+        results_ = json.load(f)
+    return results_
+
+
+def append_score(score_list: list, path_idx_list: list, score: float, idx: int):
+    score_list.append(score)
+    path_idx_list.append(idx)
+
+
+def print_best_score(score_list: list, path_idx_list: list, best_function, score_name: str, paths_: list):
+    best_score_idx = best_function(score_list)
+    best_score = score_list[best_score_idx]
+    best_path_idx = path_idx_list[best_score_idx]
+    best_path = paths_[best_path_idx]
+    print('Best {0} of {1} found at {2}'.format(score_name, best_score, best_path))
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--results-dir', type=str, default='results')
+parser.add_argument('--csv', type=str, default=None)
+args = parser.parse_args()
+
+
+if args.csv is not None:
+    df = pd.read_csv(args.csv)
+    paths = df.Path
+    total_rmse_scores = []
+    total_rmse_path_indices = []
+    mean_spearman_scores = []
+    mean_spearman_path_indices = []
+    top_k_regret_scores = []
+    top_k_regret_path_indices = []
+
+    for path_idx, results_dir in enumerate(paths):
+        results = load_results(results_dir)
+        problem_scores = results['scores']
```

Comment:
some results don't have scores if the experiment was terminated early or failed. this should handle that case